src/linear_algebra/basic.lean

```diff
@@ -2102,10 +2102,46 @@ end
 
 end field
 
+/-- Given `f : M₁ ≃ₗ[R] M₂` and `U` a submodule of `M₂`, `f.comap U` is the
+induced `linear_equiv` from `U.comap f.to_linear_map` to `U`. -/
+@[simps]
+def comap [ring R] [add_comm_group M] [module R M]
+  [add_comm_group M₂] [module R M₂] (f : M ≃ₗ[R] M₂) (U : submodule R M₂) :
+  U.comap f.to_linear_map ≃ₗ[R] U := f.of_submodules _ _
+begin
+  ext x,
+  simp_rw [submodule.mem_map, submodule.mem_comap],
+  split; intro hx,
+  { rcases hx with ⟨y, hy, rfl⟩,
+    exact hy },
+  { refine ⟨f.inv_fun x, _⟩,
+    simpa }
+end
+
 end linear_equiv
 
 namespace submodule
 
+/-- Given `p` a submodule of the module `M` and `q` a submodule of `p`, `p.equiv_subtype_map q`
+is the natural `linear_equiv` between `q` and `q.map p.subtype`. -/
+@[simps]
```

Comment:
This doesn't generate a particularly pleasant simp lemma:
```
equiv_subtype_map_apply :
  ∀ (p : submodule ?M_1 ?M_2) (q : submodule ?M_1 ↥p) (ᾰ : ↥q),
    ⇑(p.equiv_subtype_map q) ᾰ = (linear_map.cod_restrict (map p.subtype q) (p.subtype.dom_restrict q) _).to_fun ᾰ
```
Specifying `to_fun` explicitly as
```
{ to_fun := (p.subtype.dom_restrict q).cod_restrict _
    begin
      rintro ⟨x, hx⟩,
      refine ⟨x, hx, rfl⟩,
    end,
  inv_fun :=
    begin
      rintro ⟨x, hx⟩,
      refine ⟨⟨x, _⟩, _⟩;
      rcases hx with ⟨⟨_, h⟩, _, rfl⟩;
      assumption
    end,
  left_inv := λ ⟨⟨_, _⟩, _⟩, rfl,
  right_inv := λ ⟨x, ⟨_, h⟩, _, rfl⟩, rfl,
  map_add' := linear_map.map_add _,
  map_smul' := linear_map.map_smul _}
```
eliminates the `.to_fun` here.

Reply:
I think we should add the apply lemma manually here since we can get rid of `cod_restrict` while the `simps` lemma does not